### ISSUE TITLE
Improve CreationSpecsBase test hierarchy's threadsafety

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1069]
At least one test (a spec) [fails](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/build/189-fjpeidki#L751), due to what appears to be  a use of shared state:

```
      System.InvalidOperationException : Collection was modified; enumeration operation may not execute.
build.cmd :     FakeItEasy.Specs.GenericCreationSpecs.UseSuccessfulConstructor() [08] And the one-parameter constructor was not tried [FAIL]
At line:4 char:5
+     & .\build.cmd
+     ~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (    FakeItEasy....ot tried [FAIL]:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
      Stack Trace:
           at System.Collections.Generic.SortedSet`1.Enumerator.MoveNext()
           at System.Linq.Enumerable.<CastIterator>d__29`1.MoveNext()
           at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
           at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
        C:\projects\fluentassertions-vf06b\Src\Core\Collections\CollectionAssertions.cs(335,0): at FluentAssertions.Collections.CollectionAssertions`2.BeEquivalentTo[T](IEnumerable`1 expected, String because, Object[] becauseArgs)
        C:\projects\fakeiteasy\tests\FakeItEasy.Specs\CreationSpecs.cs(74,0): at FakeItEasy.Specs.CreationSpecsBase.<>c.<UseSuccessfulConstructor>b__2_6()
        C:\projects\xbehave-net\src\Xbehave.Core\StringExtensions.cs(36,0): at Xbehave.StringExtensions.<>c__DisplayClass0_0.<x>b__0(IStepContext c)
        C:\projects\xbehave-net\src\Xbehave.Execution\StepInvoker.cs(75,0): at Xbehave.Execution.StepInvoker.<>c__DisplayClass8_1.<<InvokeBodyAsync>b__1>d.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

**Comment by blairconrad** [comment]
I took a very quick gander, and `CreationSpecsBase` (and classes extending it) look like the only offender.

**Comment by blairconrad** [comment]
This issue was fixed in [Release 3.3.0](https://github.com/FakeItEasy/FakeItEasy/releases/3.3.0).
